docs/pages/includes/database-access/aws-troubleshooting.mdx

```diff
@@ -24,3 +24,139 @@ program you can verify connections to databases:
 $ nc -zv postgres-instance-1.sadas.us-east-1.rds.amazonaws.com 5432
 # Connection to postgres-instance-1.sadas.us-east-1.rds.amazonaws.com (172.31.24.172) 5432 port [tcp/postgresql] succeeded!
 ```
+
+### Not authorized to perform `sts:AssumeRole`
+
+The Database Service assumes an IAM role in one of following situations:
+
+- An IAM role is used as `db_user` when accessing AWS services that require IAM
+  roles as database users, such as DynamoDB, Keyspaces, Opensearch, and
+  Redshift Serverless.
+- `assume_role_arn` field is specified for the database resources or dynamic
+  resource matchers.
+
+<Details title="Role chaining">
+When both of the above conditions are true for a database connection, the
+Database Service performs a role chaining by assuming the IAM role specified
+`assume_role_arn` first then using that IAM role to assume the IAM role for
+`db_user`.
+</Details>
+
+You may encounter the following error if the trust relationship is not
+configured properly between the IAM roles:
+```
+AccessDenied: User: arn:aws:sts::111111111111:assumed-role/database-service-role/i-* is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::111111111111:role/database-user-role
+```
+
+To allow IAM Role `role1` to assume IAM Role `role2`, the following is
+generally required:
+
+<Details title="1. Configure Trust Relationships on role2">
+`role1` or its AWS account should be set as `Principal` in `role2`'s trust
+policy.
+<Tabs>
+<TabsItem label="Role as principal">
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::111111111111:role/role1"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+```
+</TabsItem>
+<TabsItem label="Account as principal">
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::111111111111:root"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+```
+</TabsItem>
+<TabsItem label="Cross-account with external-id">
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::123456789012:role/role1"
+      },
+      "Action": "sts:AssumeRole",
+      "Condition": {
+        "StringEquals": {
+          "sts:ExternalId": "example-external-id"
+        }
+      }
+    }
+  ]
```

Comment:
since you can use cross-account policy with either the account as principal or a specific role, can we include an example for both

Reply:
I was about to add it but then decided to keep it as it is. For cross-account, it's better to be more secure with a single role (or a list of roles) instead of trusting the account. ExternalID is for better security as well. Also, I am sure people can figure out how to trust cross-accounts using the second example.